tests/network/bindingplugin.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libkvconfig"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+const (
+	passtBindingName  = "passt"
+	passtSidecarImage = "registry:5000/kubevirt/network-passt-binding:devel"
+	passtNetAttDef    = "netbindingpasst"
+	passtType         = "cni-passt-binding-plugin"
+)
+
+var _ = SIGDescribe("[Serial]network binding plugin", Serial, decorators.NetCustomBindingPlugins, func() {
+
+	libkvconfig.WithFeatureGate(virtconfig.NetworkBindingPlugingsGate)
+
+	Context("passt", func() {
+		libkvconfig.WithNetBindingPlugin(passtBindingName, v1.InterfaceBindingPlugin{
+			SidecarImage:                passtSidecarImage,
+			NetworkAttachmentDefinition: passtNetAttDef,
+		})
```

Comment:
Is there a reason not to use existing similar functions:
https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go#L1650
https://github.com/kubevirt/kubevirt/blob/main/tests/utils.go#L1798C6-L1798C57

Reply:
I am not sure why you declare them wrong and heavy, they are used in tests for quite long time and prove themselves. 
In general we still have some flaks, but I don't recall these fn's involved.

In general changing Kubevirt config is a heavy operation, depending on the change it may trigger create/remove pods and changes to other Kubevirt components. 
Some changes needs to be verified before continuing with the tests to prevent flakes.
This is why the UpdateKubevirt.. function checks revision-numbers of Kubevirt components.

The implement this PR introduce don't do it, I assume it wont introduce flakes because it doesnt trigger chagnes for other components.

It would be nicer to have this change on it own PR where also it replaces to current implantation (at least in some places..), I recall similar changes done that way.

Reply:
The existing tooling are heavy because they fetch the KV, then changes it back using a special merge.
That special merge is not working well with our new binding `map` (it does not support merging of maps). All this instead of using a simple JSON patch to update what is needed.
Just to be clear, I did started using it and it failed when changing the map. And I did not want to change for all others this strategic merge patch type, I do not know what side effects it will have.

If we need to wait for the configuration to be distributed or not, is a fine concern.
If it is indeed needed, I guess it needs to be specific to a component and the tooling should allow to control this somehow. I have not seen this problem in the executions I ran, but if this is not safe, we could add it.